force-app/main/default/lwc/ldsCreateRecord/__tests__/ldsCreateRecord.test.js

```diff
@@ -0,0 +1,138 @@
+        });
+        document.body.appendChild(element);
+
+        const inputEl = element.shadowRoot.querySelector(
+            'lightning-input[class="slds-m-bottom_x-small"]'
+        );
+        inputEl.value = USER_INPUT;
+        inputEl.dispatchEvent(new CustomEvent('change'));
+
+        const buttonEl = element.shadowRoot.querySelector('lightning-button');
+        buttonEl.click();
+
+        // Return a promise to wait for any asynchronous DOM updates.
+        return Promise.resolve().then(() => {
+            expect(createRecord).toHaveBeenCalled();
+            expect(createRecord.mock.calls[0]).toEqual(INPUT_PARAMETERS);
+        });
+    });
+
+    it('displays an account id after record creation', () => {
+        const USER_INPUT = 'Gomez Inc.';
+        const RESULT = '0011700000r6zdbAAA';
+        createRecord.mockResolvedValue(mockCreateRecord);
+
+        const element = createElement('c-lds-create-record', {
+            is: LdsCreateRecord
+        });
+        document.body.appendChild(element);
+
+        const inputEl = element.shadowRoot.querySelector(
+            'lightning-input[class="slds-m-bottom_x-small"]'
+        );
+        inputEl.value = USER_INPUT;
+        inputEl.dispatchEvent(new CustomEvent('change'));
+
+        const buttonEl = element.shadowRoot.querySelector('lightning-button');
+        buttonEl.click();
+
+        // Return a promise to wait for any asynchronous DOM updates.
+        return flushPromises().then(() => {
+            const displayEl = element.shadowRoot.querySelector(
+                'lightning-input:not([class="slds-m-bottom_x-small"])'
+            );
+            expect(displayEl.value).toBe(RESULT);
+        });
+    });
+
+    it('displays a success toast after record creation', () => {
+        const USER_INPUT = 'Gomez Inc.';
+        createRecord.mockResolvedValue(mockCreateRecord);
+
+        const element = createElement('c-lds-create-record', {
+            is: LdsCreateRecord
+        });
+        document.body.appendChild(element);
+
+        // Mock handler for toast event
+        const handler = jest.fn();
+        // Add event listener to catch toast event
+        element.addEventListener('lightning__showtoast', handler);
+
+        const inputEl = element.shadowRoot.querySelector(
+            'lightning-input[class="slds-m-bottom_x-small"]'
+        );
+        inputEl.value = USER_INPUT;
+        inputEl.dispatchEvent(new CustomEvent('change'));
+
+        const buttonEl = element.shadowRoot.querySelector('lightning-button');
+        buttonEl.click();
+
+        // Return a promise to wait for any asynchronous DOM updates.
+        return Promise.resolve().then(() => {
+            // Check if toast event has been fired
+            expect(handler).toHaveBeenCalled();
+        });
+    });
+
+    it('displays an error toast on createRecord error', () => {
+        const USER_INPUT = 'invalid';
+        createRecord.mockResolvedValue = { id: '111' };
```

Comment:
This should reject the promise to fire the "error" toast event.

Reply:
Done, was from my tests.